File: src/org/cappuccino_project/ide/intellij/plugin/extensions/plist/ObjJPlistFileType.java
package org.cappuccino_project.ide.intellij.plugin.extensions.plist;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class ObjJPlistFileType extends LanguageFileType {
    public static final String FILE_EXTENSION = "plist";
    public static final ObjJPlistFileType INSTANCE = new ObjJPlistFileType();

    private ObjJPlistFileType() {
        super(ObjJPlistLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "Objective-J Property Plist";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "An Objective-J plist file";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return FILE_EXTENSION;
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return ObjJPlistIcons.DOCUMENT_ICON;
    }
}

File: src/main/java/cappuccino/ide/intellij/plugin/extensions/plist/ObjJPlistFileType.kt
package cappuccino.ide.intellij.plugin.extensions.plist

import com.intellij.openapi.fileTypes.LanguageFileType

import javax.swing.*

class ObjJPlistFileType private constructor() : LanguageFileType(ObjJPlistLanguage.INSTANCE) {

    override fun getName(): String {
        return "Objective-J Property Plist"
    }

    override fun getDescription(): String {
        return "An Objective-J plist file"
    }

    override fun getDefaultExtension(): String {
        return FILE_EXTENSION
    }

    override fun getIcon(): Icon? {
        return ObjJPlistIcons.DOCUMENT_ICON
    }

    companion object {
        val FILE_EXTENSION = "plist"
        val INSTANCE = ObjJPlistFileType()
    }
}
